src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

```diff
@@ -1942,9 +2089,16 @@ private void addMedInRowInInventorySearchList(MedicalInventoryRow inventoryRow)
 		}
 	}
 	private void resetVariable() {
+		validateButton.setEnabled(true);
 		inventoryRowsToDelete.clear();
 		lotsDeleted.clear();
 		inventoryRowListAdded.clear();
 		lotsSaved.clear();
 	}
+	
+	private void disableValidateButton() {
+		if (validateButton.isEnabled()) {
+			validateButton.setEnabled(false);
+		}
+	}
```

Comment:
why do we need this method in place of just having `validateButton.setEnabled(false)` where needed?

Reply:
just to disable validate  button  when some update are done to allow user save before validate.  I have create that methode to avoid many duplicated code.

Reply:
`validateButton.setEnabled(false)` is enough, to avoid one-line command you use 5 lines of code containing one method and one (useless) condition?!

Reply:
to have saving before validation have you tried to just do `saveButton.doClick()`?

Reply:
ok

Reply:
no, just when you perform validation, you also do first `saveButton.doClick()` and if it is ok, you continue with validation

Reply:
if nothing has been update you will have information message "This inventory is already saved. Please enter at least one inventory row before continue."

Reply:
Hello I have apply you suggestion